sdk/eventhub/Azure.Messaging.EventHubs/design/api-skeleton-january-2020.md

```diff
@@ -0,0 +1,535 @@
+    
+    public virtual ValueTask<EventDataBatch> CreateBatchAsync(CreateBatchOptions options, CancellationToken cancellationToken = default);
+    public virtual ValueTask<EventDataBatch> CreateBatchAsync(CancellationToken cancellationToken = default);
+    
+    public virtual Task SendAsync(EventDataBatch eventBatch, CancellationToken cancellationToken = default);
+}
+
+public class EventHubProducerClientOptions 
+{
+    public EventHubConnectionOptions ConnectionOptions { get; set; }
+    public EventHubsRetryOptions RetryOptions { get; set; }
+}
+
+public class CreateBatchOptions
+{
+    public long? MaximumSizeInBytes { get; set; }
+    public string PartitionId { get; set; }
+    public string PartitionKey { get; set; }
+}
+
+public sealed class EventDataBatch : IDisposable 
+{
+    public int Count { get; }
+    public long MaximumSizeInBytes { get; }
+    public long SizeInBytes { get; }
+    public void Dispose();
+    public bool TryAdd(EventData eventData);
+}
+```
+
+### `Azure.Messaging.EventHubs.Processor`
+```csharp
+public struct ProcessEventArgs 
+{
+    public bool HasEvent { get; }
+    public PartitionContext Partition { get; }
+    public EventData Data { get; }
+    public CancellationToken CancellationToken { get; }
+    
+    public ProcessEventArgs(
+        PartitionContext partition,
+        EventData data,
+        Func<Task> updateCheckpointImplementation,
+        CancellationToken cancellationToken = default);
+    
+    public Task UpdateCheckpointAsync(CancellationToken cancellationToken = default);
+}
+
+public class PartitionInitializingEventArgs
+{
+    public string PartitionId { get; }
+    public EventPosition DefaultStartingPosition { get; set; }
+    public CancellationToken CancellationToken { get; }
+    
+    public PartitionInitializingEventArgs(
+        string partitionId,
+        EventPosition defaultStartingPosition,
+        CancellationToken cancellationToken = default);
+}
+
+public class PartitionClosingEventArgs 
+{
+    public string PartitionId { get; }
+    public ProcessingStoppedReason Reason { get; }
+    public CancellationToken CancellationToken { get; }
+    
+    public PartitionClosingEventArgs(
+        string partitionId,
+        ProcessingStoppedReason reason,
+        CancellationToken cancellationToken = default);
+}
+
+public struct ProcessErrorEventArgs
+{
+    public string PartitionId { get; }
+    public string Operation { get; }
+    public Exception Exception { get; }
+    public CancellationToken CancellationToken { get; }
+    
+    public ProcessErrorEventArgs(
+        string partitionId,
+        string operation,
+        Exception exception,
+        CancellationToken cancellationToken = default);
+}
+
+public enum ProcessingStoppedReason 
+{
+    Shutdown,
+    OwnershipLost
+}
+```
+
+## Package: Azure.Messaging.EventHubs.Processor
```

Comment:
If we decide to remove above header, then we should remove this one as well.

Reply:
This header corresponds to the package, the previous to a namespace in a different package;  they're both needed.